.github/workflows/release.yml

```diff
@@ -412,3 +412,92 @@ jobs:
           tags: ${{ steps.docker_meta.outputs.tags }}
           labels: ${{ steps.docker_meta.outputs.labels }}
           platforms: linux/amd64,linux/arm64
+
+  release-dashmate:
+    name: Release Dashmate packages
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-22.04, macos-12]
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v3
+
+      - name: Setup Node.JS
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16'
+
+      - name: Install macOS build deps
+        if: runner.os == 'macOS'
+        run: |
+          brew install llvm docker colima
+          colima start
+          echo "/usr/local/opt/llvm/bin" >> $GITHUB_PATH
+
+      - name: Install Linux build deps
+        if: runner.os == 'Linux'
+        run: sudo apt-get install -y nsis
+
+      - uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: stable
+          target: wasm32-unknown-unknown
+
+      - name: Enable corepack
+        run: corepack enable
+
+      - name: Disable NPM audit
+        run: npm config set audit false
+
+      - name: Enable Yarn unplugged modules cache
+        uses: actions/cache@v3
+        with:
+          path: '.yarn/unplugged'
+          key: ${{ runner.os }}-yarn-unplugged-${{ hashFiles('yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-unplugged-
+
+      # Workaround for lack of yarn 3 support in oclif
+      - name: Patch package-json
```

Comment:
Didn't you fix it with the patch?

Reply:
It is [not possible](https://github.com/yarnpkg/berry/issues/922) to use a patch to add dependencies to a project, because the dependency tree is resolved before patches are applied. Since oclif does not play well with monorepos or yarn, we need to add these dependencies to the root package, so a hotfix like this and `YARN_ENABLE_IMMUTABLE_INSTALLS: false` to allow changes to `yarn.lock` is needed.

Reply:
I think this is a horrible hack though. But it's the only way I could find that worked, maybe it is because we are also attempting to patch the same package that is missing the dependency. I was unable to get it working by declaring a `packageExtension` in `yarnrc.yml`, which would be the normal solution for this case. It always throws errors like this:
```
Error: Your application tried to access lodash, but it isn't declared in your dependencies; this makes the require call ambiguous and unsound.
Required package: lodash
Required by: /home/strophy/Code/strophy/platform/.yarn/cache/oclif-patch-5d6a64e939-49ca67dd4f.zip/node_modules/oclif/lib/
Require stack:
- /home/strophy/Code/strophy/platform/.yarn/cache/oclif-patch-5d6a64e939-49ca67dd4f.zip/node_modules/oclif/lib/upload-util.js
- /home/strophy/Code/strophy/platform/.yarn/cache/oclif-patch-5d6a64e939-49ca67dd4f.zip/node_modules/oclif/lib/tarballs/build.js
- /home/strophy/Code/strophy/platform/.yarn/cache/oclif-patch-5d6a64e939-49ca67dd4f.zip/node_modules/oclif/lib/tarballs/index.js
- /home/strophy/Code/strophy/platform/.yarn/cache/oclif-patch-5d6a64e939-49ca67dd4f.zip/node_modules/oclif/lib/commands/pack/tarballs.js
- /home/strophy/Code/strophy/platform/.yarn/cache/@oclif-core-npm-1.21.0-d51f547b63-31fa68afb7.zip/node_modules/@oclif/core/lib/module-loader.js
- /home/strophy/Code/strophy/platform/.yarn/cache/@oclif-core-npm-1.21.0-d51f547b63-31fa68afb7.zip/node_modules/@oclif/core/lib/config/plugin.js
- /home/strophy/Code/strophy/platform/.yarn/cache/@oclif-core-npm-1.21.0-d51f547b63-31fa68afb7.zip/node_modules/@oclif/core/lib/config/config.js
- /home/strophy/Code/strophy/platform/.yarn/cache/@oclif-core-npm-1.21.0-d51f547b63-31fa68afb7.zip/node_modules/@oclif/core/lib/config/index.js
- /home/strophy/Code/strophy/platform/.yarn/cache/@oclif-core-npm-1.21.0-d51f547b63-31fa68afb7.zip/node_modules/@oclif/core/lib/command.js
- /home/strophy/Code/strophy/platform/.yarn/cache/@oclif-core-npm-1.21.0-d51f547b63-31fa68afb7.zip/node_modules/@oclif/core/lib/index.js
- /home/strophy/Code/strophy/platform/.yarn/cache/oclif-patch-5d6a64e939-49ca67dd4f.zip/node_modules/oclif/bin/run
Code: MODULE_NOT_FOUND
```

It's super hard to understand, I can't figure out which module in the stack is missing the dependency. The only way I found to resolve the error is to temporarily declare the missing deps in the root `package.json`, have you had this problem before?